packages/arcgis-rest-request/src/ApiKeyManager.ts

```diff
@@ -52,11 +55,77 @@ export class ApiKeyManager
   }
 
   /**
-   * Gets a token (the API Key).
+   * Gets the current access token (the API Key).
+   */
+  get token() {
+    return this.key;
+  }
+
+  /**
+   * Gets the current access token (the API Key).
    */
   public getToken(url: string) {
     return Promise.resolve(this.key);
   }
+
+  /**
+   * Converts the `ApiKeyManager` instance to a JSON object. This is called when the instance is serialized to JSON with [`JSON.stringify()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify).
+   *
+   * ```js
+   * import { ApiKeyManager } from '@esri/arcgis-rest-request';
+   *
+   * const apiKey = new ApiKeyManager.fromKey("...")
+   *
+   * const json = JSON.stringify(session);
+   * ```
+   *
+   * @returns A plain object representation of the instance.
+   */
+  toJSON() {
+    return {
+      type: "ApiKeyManager",
+      token: this.key,
+      username: this.username,
+      portal: this.portal
+    };
+  }
+
+  /**
+   * Serializes the ApiKeyManager instance to a JSON string.
+   *
+   * ```js
+   * import { ApiKeyManager } from '@esri/arcgis-rest-request';
+   *
+   * const apiKey = new ApiKeyManager.fromKey("...")
+   *
+   * localStorage.setItem("apiKey", apiKey.serialize());
+   * ```
+   * @returns {string} The serialized JSON string.
+   */
+  serialize() {
+    return JSON.stringify(this);
```

Comment:
I think you meant for this to be 
```suggestion
    return JSON.stringify(this.toJSON());
```